sim/rogue/combat/killing_spree.go

```diff
@@ -79,29 +79,33 @@ func (rogue *Rogue) registerKillingSpreeSpell() {
 			},
 			IgnoreHaste: true,
 			CD: core.Cooldown{
-				Timer:    rogue.NewTimer(),
-				Duration: time.Minute*2 - core.TernaryDuration(rogue.HasMajorGlyph(proto.RogueMajorGlyph_GlyphOfKillingSpree), time.Second*45, 0),
+				Timer:    comRogue.NewTimer(),
+				Duration: time.Minute * 2,
 			},
 		},
 
 		ApplyEffects: func(sim *core.Simulation, u *core.Unit, s2 *core.Spell) {
-			rogue.BreakStealth(sim)
-			rogue.KillingSpreeAura.Activate(sim)
+			comRogue.BreakStealth(sim)
+			comRogue.KillingSpreeAura.Activate(sim)
 		},
 	})
 
-	rogue.AddMajorCooldown(core.MajorCooldown{
-		Spell:    killingSpreeSpell,
+	comRogue.AddMajorCooldown(core.MajorCooldown{
+		Spell:    comRogue.KillingSpree,
 		Type:     core.CooldownTypeDPS,
-		Priority: core.CooldownPriorityLow,
+		Priority: core.CooldownPriorityDefault,
 		ShouldActivate: func(sim *core.Simulation, c *core.Character) bool {
-			if bf := rogue.GetMajorCooldown(BladeFlurryActionID); bf != nil && bf.IsReady(sim) {
-				return false
-			}
-			if rogue.CurrentEnergy() > 60 || (rogue.CurrentEnergy() > 30 && rogue.AdrenalineRushAura.IsActive()) {
+			if comRogue.CurrentEnergy() > 40 || comRogue.AdrenalineRushAura.IsActive() {
 				return false
 			}
 			return true
 		},
 	})
 }
+
+func (comRogue *CombatRogue) registerKillingSpreeCD() {
+	if !comRogue.Talents.KillingSpree {
+		return
+	}
+	comRogue.registerKillingSpreeSpell()
+}
```

Comment:
Unneccesery function. Just move the check to the above function and call it directly